src/omv/modules/py_image.c

```diff
@@ -6309,97 +6373,68 @@ mp_obj_t py_image_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw
-
-        image.w = mp_obj_get_int(shape[1]);
-        PY_ASSERT_TRUE_MSG(image.w > 0, "Image width must be > 0");
-
-        int channels = mp_obj_get_int(shape[2]);
+    #if defined(MODULE_ULAB_ENABLED)
+    } else if (MP_OBJ_IS_TYPE(args[ARG_arg].u_obj, &ulab_ndarray_type)) {
+        ndarray_obj_t *array = MP_OBJ_TO_PTR(args[ARG_arg].u_obj);
 
-        if (channels == 1) {
-            image.pixfmt = PIXFORMAT_GRAYSCALE;
-        } else if (channels == 3) {
-            image.pixfmt = PIXFORMAT_RGB565;
-        } else {
-            mp_raise_ValueError(MP_ERROR_TEXT("Channels must be 1 or 3"));
+        if (array->dtype != NDARRAY_FLOAT) {
+            mp_raise_msg(&mp_type_ValueError, MP_ERROR_TEXT("Expected a ndarray with dtype float"));
         }
 
-        mp_obj_t *strides;
-        mp_obj_get_array_fixed_n(args[ARG_strides].u_obj, 2, &strides);
-
-        int start = 0;
-        int start_r = 0;
-        int start_g = 0;
-        int start_b = 0;
-
-        if (channels == 1) {
-            start = mp_obj_get_int(strides[0]);
-            PY_ASSERT_TRUE_MSG(start >= 0, "Start must be >= 0");
-        } else {
-            mp_obj_t *rgb_strides;
-            mp_obj_get_array_fixed_n(strides[0], 3, &rgb_strides);
-
-            start_r = mp_obj_get_int(rgb_strides[0]);
-            PY_ASSERT_TRUE_MSG(start_r >= 0, "R Start must be >= 0");
-
-            start_g = mp_obj_get_int(rgb_strides[1]);
-            PY_ASSERT_TRUE_MSG(start_g >= 0, "G Start must be >= 0");
-
-            start_b = mp_obj_get_int(rgb_strides[2]);
-            PY_ASSERT_TRUE_MSG(start_b >= 0, "B Start must be >= 0");
+        if (!((array->ndim == 2) || ((array->ndim == 3) && (array->shape[ULAB_MAX_DIMS - 1] == 3)))) {
+            mp_raise_msg(&mp_type_ValueError,
+                         MP_ERROR_TEXT("Expected a ndarray with shape (height, width) or (height, width, 3"));
         }
 
-        int step = mp_obj_get_int(strides[1]);
-        PY_ASSERT_TRUE_MSG(step > 0, "Step must be > 0");
-
-        mp_obj_t *items;
-        size_t items_len;
-        mp_obj_get_array(args[ARG_arg].u_obj, &items_len, &items);
-
-        int size = image.w * image.h;
-        int step_max = (size - 1) * step;
-
-        if (channels == 1) {
-            if (items_len <= (start + step_max)) {
-                mp_raise_ValueError(MP_ERROR_TEXT("Array too small"));
-            }
+        if (array->ndim == 2) {
+            image.w = array->shape[ULAB_MAX_DIMS - 1];
+            image.h = array->shape[ULAB_MAX_DIMS - 2];
+            image.pixfmt = PIXFORMAT_GRAYSCALE;
         } else {
-            if ((items_len <= (start_r + step_max)) ||
-                (items_len <= (start_g + step_max)) ||
-                (items_len <= (start_b + step_max))) {
-                mp_raise_ValueError(MP_ERROR_TEXT("Array too small"));
-            }
+            image.w = array->shape[ULAB_MAX_DIMS - 2];
+            image.h = array->shape[ULAB_MAX_DIMS - 3];
```

Comment:
```suggestion
            image.h = array->shape[ULAB_MAX_DIMS - 3];
```
This code is full of assumptions about the DIMS, it's 4 for all boards but what if we reduced it again for any reason ? At least check for the right number of DIMS.